packages/react/src/components/Tree/TreeItem.tsx

```diff
@@ -0,0 +1,240 @@
+  title?: ShorthandValue<TreeTitleProps>
+}
+
+class TreeItem extends UIComponent<WithAsProp<TreeItemProps>> {
+  static create: Function
+
+  static displayName = 'TreeItem'
+
+  static className = 'ui-tree__item'
+
+  static slotClassNames: TreeItemSlotClassNames = {
+    title: `${TreeItem.className}__title`,
+    subtree: `${TreeItem.className}__subtree`,
+  }
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      content: false,
+    }),
+    id: PropTypes.string,
+    items: customPropTypes.collectionShorthand,
+    index: PropTypes.number,
+    exclusive: PropTypes.bool,
+    level: PropTypes.number,
+    onTitleClick: PropTypes.func,
+    onFocusFirstChild: PropTypes.func,
+    onFocusParent: PropTypes.func,
+    onSiblingsExpand: PropTypes.func,
+    open: PropTypes.bool,
+    parent: customPropTypes.itemShorthand,
+    renderItemTitle: PropTypes.func,
+    siblings: customPropTypes.collectionShorthand,
+    treeItemRtlAttributes: PropTypes.func,
+    title: customPropTypes.itemShorthand,
+  }
+
+  static defaultProps = {
+    as: 'div',
+    accessibility: treeItemBehavior,
+  }
+
+  itemRef = React.createRef<HTMLElement>()
+  treeRef = React.createRef<HTMLElement>()
+
+  actionHandlers = {
+    performClick: e => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      this.handleTitleClick(e)
+    },
+    focusParent: e => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      this.handleParentFocus(e)
+    },
+    collapse: e => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      this.handleTitleClick(e)
+    },
+    expand: e => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      this.handleTitleClick(e)
+    },
+    focusFirstChild: e => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      this.handleFocusFirstChild(e)
+    },
+    expandSiblings: e => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      this.handleSiblingsExpand(e)
+    },
+  }
+
+  eventComesFromChildItem = e => {
+    return e.currentTarget !== e.target
+  }
+
+  handleTitleClick = e => {
+    _.invoke(this.props, 'onTitleClick', e, this.props)
+  }
+
+  handleParentFocus = e => {
+    _.invoke(this.props, 'onFocusParent', e, this.props)
+  }
+
+  handleFocusFirstChild = e => {
+    _.invoke(this.props, 'onFocusFirstChild', e, this.props)
+  }
+
+  handleSiblingsExpand = e => {
+    _.invoke(this.props, 'onSiblingsExpand', e, this.props)
+  }
+
+  handleTitleOverrides = (predefinedProps: TreeTitleProps) => ({
+    onClick: (e, titleProps) => {
+      this.handleTitleClick(e)
+      _.invoke(predefinedProps, 'onClick', e, titleProps)
+    },
+  })
+
+  renderContent() {
+    const { items, title, renderItemTitle, open, level, siblings, index } = this.props
+    const hasSubtree = !_.isNil(items)
+
+    return TreeTitle.create(title, {
+      defaultProps: {
+        className: TreeItem.slotClassNames.title,
+        open,
+        hasSubtree,
+        as: hasSubtree ? 'span' : 'a',
```

Comment:
we can consider to have it as part of `treeTitleBehavior` to define if render `span` or `a`